gtk3/sample/misc/gtkglarea2.rb

```diff
@@ -0,0 +1,192 @@
+=begin
+  gtkglarea2.rb draw a triangle using shaders on a grey background
+  this script need the opengl-bindings gem from :
+  https://github.com/vaiorabbit/ruby-opengl
+
+  gem install opengl-bindings
+
+  sources:
+  http://antongerdelan.net/opengl/hellotriangle.html
+  opengl-bindings/sample/RedBook/varray
+  opengl-bindings/sample/GLES/gles.rb
+  http://www.opengl-tutorial.org/beginners-tutorials/tutorial-2-the-first-triangle/
+  https://developer.gnome.org/gtk3/stable/GtkGLArea.html
+  https://www.bassi.io/articles/2015/02/17/using-opengl-with-gtk/
+  http://stackoverflow.com/questions/30337845/gldrawarrays-not-working-using-gtkglarea-in-gtk3
+=end
+
+require "gtk3"
+require "opengl"
+
+unless Gtk::Version.or_later?(3, 16, 0)
+  puts "This sample requires GTK+ 3.16.0 or later: #{Gtk::Version::STRING}"
+  exit
+end
+
+case OpenGL.get_platform
+when :OPENGL_PLATFORM_WINDOWS
+  OpenGL.load_lib("opengl32.dll", "C:/Windows/System32")
+when :OPENGL_PLATFORM_MACOSX
+  OpenGL.load_lib("libGL.dylib", "/System/Library/Frameworks/OpenGL.framework/Libraries")
+when :OPENGL_PLATFORM_LINUX
+  OpenGL.load_lib("libGL.so", "/usr/lib")
```

Comment:
Unfortunately, some linux distributions put this at different places.

e.g)
Ubuntu Linux: /usr/lib/x86_64-linux-gnu/libGL.so
CentOS amd64: /usr/lib64/libGL.so
 
In more detail, please refer to this article: http://www.pilotlogic.com/sitejoom/index.php/wiki?id=398.